File: subroutines/define-and-call/run_quiz.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0
# https://isaaccomputerscience.org/concepts/prog_sub_define_and_call

def run_quiz():
    """Ask a quiz question"""
    print("What is the capital city of Botswana?")
    answer = input()


def main():
    name = input("What is your name? ")
    run_quiz() 
    print("End of the quiz")


if __name__ == '__main__':
    main()

File: subroutines/define-and-call/run_quiz.cs
/*
Note: This file is designed to be copied out and compiled on your machine. 
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file. 
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Compile the program
4. Run the program
*/

using System;


namespace Subroutines
{
    class DefineAndCall
    {
        // The Main method is the entry point for all C# programs
        public static void Main(string[] args) {
            Console.Write("What is your name? ");
            string name = Console.ReadLine();
            RunQuiz();
            Console.WriteLine("End of the quiz");
        }
        

        // Simulates a quiz
        public static void RunQuiz() {
            Console.WriteLine("What is the capital city of Botswana?");
            string answer = Console.ReadLine();
        }


    }
}
